src/secret_base.c

```diff
@@ -0,0 +1,2272 @@
+            case 2:
+                mixers[0].records = records + 3 * recordSize;
+                mixers[0].version = gLinkPlayers[3].version & 0xFF;
+                mixers[0].language = gLinkPlayers[3].language;
+                mixers[1].records = records + 0 * recordSize;
+                mixers[1].version = gLinkPlayers[0].version & 0xFF;
+                mixers[1].language = gLinkPlayers[0].language;
+                mixers[2].records = records + 1 * recordSize;
+                mixers[2].version = gLinkPlayers[1].version & 0xFF;
+                mixers[2].language = gLinkPlayers[1].language;
+                break;
+            case 3:
+                mixers[0].records = records + 0 * recordSize;
+                mixers[0].version = gLinkPlayers[0].version & 0xFF;
+                mixers[0].language = gLinkPlayers[0].language;
+                mixers[1].records = records + 1 * recordSize;
+                mixers[1].version = gLinkPlayers[1].version & 0xFF;
+                mixers[1].language = gLinkPlayers[1].language;
+                mixers[2].records = records + 2 * recordSize;
+                mixers[2].version = gLinkPlayers[2].version & 0xFF;
+                mixers[2].language = gLinkPlayers[2].language;
+                break;
+        }
+        sub_80EAEF4(mixers);
+        for (i = 1; i < 20; i ++)
+        {
+            if (gSaveBlock1Ptr->secretBases[i].sbr_field_1_0 == 1)
+            {
+                gSaveBlock1Ptr->secretBases[i].sbr_field_1_6 = 1;
+                gSaveBlock1Ptr->secretBases[i].sbr_field_1_0 = 0;
+            }
+        }
+        sub_80EAAF4();
+        for (i = 1; i < 20; i ++)
+        {
+            if (gSaveBlock1Ptr->secretBases[i].sbr_field_1_6 == 2)
+            {
+                gSaveBlock1Ptr->secretBases[i].sbr_field_1_6 = 0;
+            }
+        }
+        if (gSaveBlock1Ptr->secretBases[0].secretBaseId != 0 && gSaveBlock1Ptr->secretBases[0].sbr_field_e != 0xFFFF)
+        {
+            gSaveBlock1Ptr->secretBases[0].sbr_field_e ++;
+        }
+    }
+}
+
+void sub_80EB18C(struct SecretBaseRecord *bases)
+{
+    u32 i;
+
+    for (i = 0; i < 20; i ++)
+    {
+        if (bases[i].language == LANGUAGE_JAPANESE)
+        {
+            ClearSecretBase(&bases[i]);
+        }
+    }
+}
+
+void sub_80EB1AC(void)
+{
+    VarSet(VAR_0x40EC, 0);
+    VarSet(VAR_0x40ED, 0);
+    VarSet(VAR_0x40EE, 0);
+    VarSet(VAR_0x40EF, 0);
+    if (VarGet(VAR_0x4054) != 0)
+    {
+        VarSet(VAR_0x40F0, TRUE);
+    }
+    else
+    {
+        VarSet(VAR_0x40F0, FALSE);
+    }
+    gUnknown_0203A01D = FALSE;
+}
+
+void sub_80EB218(void)
+{
+    if (VarGet(VAR_0x40F0) && gUnknown_0203A01D == TRUE && !CurrentMapIsSecretBase())
+    {
+        VarSet(VAR_0x40F0, FALSE);
+        gUnknown_0203A01D = FALSE;
+        sub_80EEA70();
+        VarSet(VAR_0x40EC, 0);
+        VarSet(VAR_0x40ED, 0);
+        VarSet(VAR_0x40EE, 0);
+        VarSet(VAR_0x40EF, 0);
+        VarSet(VAR_0x40F0, FALSE);
```

Comment:
Why are some of these set to 0 and some set to FALSE? This should be consistent.

Reply:
I'll double-check, but it's intended to reflect the difference between whether the var is ised as a u16 or as a bool16.